crates/ink/src/env_access.rs

```diff
@@ -396,12 +397,26 @@ where
         ink_env::minimum_balance::<E>()
     }
 
-    /// Emits an event.
+    /// Emits an event using the "default" ABI for interactions for the ink! project.
```

Comment:
Could you document the behavior for `abi = "all"` and add a test for it?

Reply:
There's a couple of lines just after this one i.e.
```rust
/// The "default" ABI for interactions is "ink", unless the ABI is set to "sol"
/// in the ink! project's manifest file (i.e. `Cargo.toml`).
```

So, at the moment (i.e. decision not final), in "all" ABI mode, this will emit an ink! ABI encoded event.
But may be the current doc isn't clear because it doesn't mention the "all" ABI mode explicitly?

In general, I plan to put more effort into this type of doc during the final pass for deciding "default" behavior and public interface for this and similar utilities in "all" ABI mode.

I'm gonna add an integration test for the current behavior though.

Reply:
Yeah I read those lines, it's what got me to comment this here.

Without knowing more about the implementation my expectation as a dev would have been that for `abi = "all"`, the events are emitted in both SCALE and Solidity encoding. Is that also your intuition?

Reply:
Yeah, the intuition tracks, I'll change the behavior to emitting both.
Probably something to revisit for a final decision before release, or when we switch to "all" as the "default" ABI, will add a note.